### PR TITLE
Shortcut to jump to mapped location

### PR DESCRIPTION
**Pull request by GarethSharpe**
Associated Issue: #4062 

Here's the Pull Request Doc
https://devtools-html.github.io/debugger.html/CONTRIBUTING.html#pull-requests

### Summary of Changes

* Created mouse click event listener for codeMirror
* Created onClick(e: MouseEvent) function shortcut to jump to source location

### Test Plan

- [x] When click + metaKey + altKey, jump to source location

Here's the Debugger's Testing doc
https://docs.google.com/document/d/1oBMRxV8A2ag2t22YsQOxTdEv0mXKzIg0tggJjRkU1S0/edit#.
Feel free to improve it!

### Screenshots/Videos

debugger - mozilla firefox - jump to context

### CODE REVIEW

File: src/components/Editor/index.js [hunk 1]
```diff
@@ -417,6 +418,25 @@ class Editor extends PureComponent {
...
+        top: e.clientY
+      });
+
+      const sourceLocation = {
+        sourceId: selectedLocation.sourceId,
+        line: line + 1,
+        column: ch + 1
+      };
+
+      jumpToMappedLocation(sourceLocation);
```
Comment (jasonLaster) [inline src/components/Editor/index.js]: lets make a small util: `getSourceLocationFromMouseEvent` in `utils/editor/index.js`

```diff
from getSourceLocationFromMouseEvent(event) {
+       const { line, ch } = this.state.editor.codeMirror.coordsChar({
+         left: e.clientX,
+         top: e.clientY
+       });
+ 
+       return{
+         sourceId: selectedLocation.sourceId,
+         line: line + 1,
+         column: ch + 1
+       };
+ 
}
```

Comment (jasonLaster) [inline src/components/Editor/index.js]: we can use this util to simplify EditorMenu as well

Comment (jasonLaster) [inline src/components/Editor/index.js]: looks like meta should map to windows

https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/metaKey

File: src/components/Editor/index.js [hunk 2]
```diff
@@ -417,6 +419,20 @@ class Editor extends PureComponent {
...
   }
 
+  onClick(e: MouseEvent) {
+    const { selectedLocation, jumpToMappedLocation } = this.props;
+
+    if (e.metaKey && e.altKey) {
+      const { line, ch } = this.state.editor.codeMirror.coordsChar({
+        left: e.clientX,
+        top: e.clientY
+      });
```
Comment (jasonLaster) [inline src/components/Editor/index.js]: lets :bomb: this

**Comment by jasonLaster** [comment]
I looked quickly and found one small bug. I think the issue with the test is legit. If you right click in the gutter you should get a different context menu than the editor.

e.g. right click in the gutter should show "add conditional breakpoint". This change breaks that.

**Comment by GarethSharpe** [comment]
Thanks - I'll take a look.

**Comment by jasonLaster** [comment]
oh nice! yeah, i'll test it locally

**Pull request merged**